gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -171,30 +309,57 @@ public void close()
   }
 
   /**
+   * Get new records off the record queue and process them.
+   */
+  private void processRecords() throws IOException, DataConversionException {
+    while (true) {
+      Object record;
+      try {
+        record = this.recordQueue.get();
+      } catch (InterruptedException ie) {
+        this.logger.warn("Interrupted while trying to get a record off the queue", ie);
```

Comment:
Swallowing an InterruptedException is rarely a good idea. Getting that exception generally means we are supposed to stop doing whatever we are doing.